streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -798,6 +806,12 @@ void runOnce() {
                         final int committed = taskManager.maybeCommitActiveTasksPerUserRequested();
 
                         if (committed > 0) {
+                            if (eosBetaEnabled) {
+                                taskManager.commitAllActive();
+                                producer.commitTransaction();
+                                producer.beginTransaction();
```

Comment:
prop: do you think we should name this as `threadProducer` for readability? cc @guozhangwang

Reply:
We can, but it's a member of `StreamThread` so it should be obviously the thread-producer IMHO.